hack/e2e-suite/guestbook.sh

```diff
@@ -38,30 +38,78 @@ GUESTBOOK="${KUBE_ROOT}/examples/guestbook"
 function teardown() {
   ${KUBECTL} stop -f "${GUESTBOOK}"
   if [[ "${KUBERNETES_PROVIDER}" == "gce" ]]; then
-    gcloud compute forwarding-rules delete "${INSTANCE_PREFIX}-default-frontend" || true
-    gcloud compute target-pools delete "${INSTANCE_PREFIX}-default-frontend" || true
+    gcloud compute forwarding-rules delete "${INSTANCE_PREFIX}-default-frontend" --region="${ZONE}" -q || true
+    gcloud compute target-pools delete "${INSTANCE_PREFIX}-default-frontend" --region="${ZONE}" -q || true
+    gcloud compute firewall-rules delete guestbook-e2e-minion-8000 -q || true
   fi  
 }
 
+function wait_for_running() {
+  echo "Waiting for pods to come up."
+  frontends=$(${KUBECTL} get pods -l name=frontend -o template '--template={{range.items}}{{.id}} {{end}}')
+  master=$(${KUBECTL} get pods -l name=redis-master -o template '--template={{range.items}}{{.id}} {{end}}')
+  slaves=$(${KUBECTL} get pods -l name=redisslave -o template '--template={{range.items}}{{.id}} {{end}}')
+  pods=$(echo $frontends $master $slaves)
+
+  all_running=0
+  for i in $(seq 1 30); do
+    sleep 10
+    all_running=1
+    for pod in $pods; do
+      status=$(${KUBECTL} get pods $pod -o template '--template={{.currentState.status}}') || true
```

Comment:
Double quotes around variables that are supposed to expand to a single token, in this case `"${pod}"`.

We also recommend using `${...}` around variable names (except for single character special variables like `$@`, `$1`, etc.)

https://google-styleguide.googlecode.com/svn/trunk/shell.xml?showone=Variable_expansion#Variable_expansion